modules/passkey/test/userStories/createSafeWithPasskeySignerAsOwner.ts

```diff
@@ -0,0 +1,168 @@
+
+    const navigator = {
+      credentials: new WebAuthnCredentials(),
+    }
+
+    // Create a WebAuthn credential for the signer
+    const credential = navigator.credentials.create({
+      publicKey: {
+        rp: {
+          name: 'Safe',
+          id: 'safe.global',
+        },
+        user: {
+          id: ethers.getBytes(ethers.id('chucknorris')),
+          name: 'chucknorris',
+          displayName: 'Chuck Norris',
+        },
+        challenge: ethers.toBeArray(Date.now()),
+        pubKeyCredParams: [{ type: 'public-key', alg: -7 }],
+      },
+    })
+
+    return {
+      user,
+      proxyFactory,
+      safeModuleSetup,
+      module,
+      entryPoint,
+      singleton,
+      signerFactory,
+      navigator,
+      verifier,
+      SafeL2,
+      credential,
+    }
+  })
+
+  it('should execute a userOp with WebAuthn signer as owner', async () => {
+    // Step 1: Setup the contracts
+    const { user, proxyFactory, safeModuleSetup, module, entryPoint, singleton, navigator, SafeL2, credential, signerFactory, verifier } =
+      await setupTests()
+
+    // Deploy a signer contract
+    const publicKey = decodePublicKey(credential.response)
+    // Deploy signer contract
+    await signerFactory.createSigner(publicKey.x, publicKey.y, await verifier.getAddress())
+    // Get signer address
+    const signer = await signerFactory.getSigner(publicKey.x, publicKey.y, await verifier.getAddress())
+
+    // Step 2: Create a userOp, sign it with Passkey signer.
+
+    // The initializer data to enable the Safe4337Module as a module on a Safe
+    const initializer = safeModuleSetup.interface.encodeFunctionData('enableModules', [[module.target]])
+
+    // Create setup data to deploy a Safe with EOA and passkey signer as owners, threshold 1, Safe4337Module as module and fallback handler
+    const setupData = singleton.interface.encodeFunctionData('setup', [
+      [signer],
+      1n,
+      safeModuleSetup.target,
+      initializer,
+      module.target,
+      ethers.ZeroAddress,
+      0,
+      ethers.ZeroAddress,
+    ])
+
+    // Predict the Safe address to construct the userOp, generate
+    const safeSalt = Date.now()
+    const safe = await proxyFactory.createProxyWithNonce.staticCall(singleton, setupData, safeSalt)
+
+    // Deploy data required in the initCode of the userOp
+    const deployData = proxyFactory.interface.encodeFunctionData('createProxyWithNonce', [singleton.target, setupData, safeSalt])
+
+    const safeOp = buildSafeUserOpTransaction(
+      safe,
+      ethers.ZeroAddress,
+      0,
+      '0x',
+      await entryPoint.getNonce(safe, 0),
+      await entryPoint.getAddress(),
+      false,
+      true,
+      {
+        initCode: ethers.solidityPacked(['address', 'bytes'], [proxyFactory.target, deployData]),
+        // Set a higher verificationGasLimit to avoid error "AA26 over verificationGasLimit"
+        verificationGasLimit: 700000,
```

Comment:
This seems overly high. I think 350000 should be enough.

Reply:
Tested with different values. Tests fails with 350k. Actually the gas usage is not the same for every run. Fails sometimes even with 570000.

Reply:
Oh wow, thats more than I would have thought...